Sprint_7

NooobStyle opened the pull request:
Sprint 7 solution

NooobStyle committed:
add allure report

NooobStyle committed:
Added Sprint_7 solution

pomegranate282922 commented on src/test/java/CreateCourierTests.java:
```diff
@@ -0,0 +1,64 @@
+
+    @BeforeEach
+    public void setUp() {
+        RestAssured.baseURI = "https://qa-scooter.praktikum-services.ru";
```
Base uri присваиваем один раз в проекте. Можно использовать класс предок или request specification в RestAssured 
Нужно исправить

pomegranate282922 commented on src/test/java/CreateCourierTests.java:
```diff
@@ -0,0 +1,64 @@
+    @Test
+    @DisplayName("Сheck successful account creation")
+    public void checkSuccessAccountCreate(){
+        String json = "{\"login\": \"test499\", \"password\": \"321321\", \"firstName\": \"Test\"}";
```
Тело запроса передаем через объект 
Нужно исправить

pomegranate282922 commented on src/test/java/CreateCourierTests.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    @Step("Assert POST /api/v1/courier response")
+    public <T> void createUserAssert(Response response, String jsonKey, T jsonValue, int expectedStatus){
```
Это выносим из тестового класса 
Нужно исправить

pomegranate282922 commented on src/test/java/utils/BasicActions.java:
```diff
@@ -0,0 +1,51 @@
+        return given()
+                .header("Content-type", "application/json")
+                .get("/api/v1/orders");
+    }
```
Отлично что вызовы ручек вынесены из тестовых классов

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Применена параметризация.
Замечания:
1. Base uri присваиваем один раз в проекте 
2. Тело запроса передаем через объект 
3. Выносим все шаги из тестовых классов

NooobStyle committed:
- Test steps have been removed from test classes
- Request bodies are passed via objects
- Base URI is passed via RequestSpecification

pomegranate282922 commented on src/test/java/utils/CreateUserData.java:
```diff
@@ -0,0 +1,52 @@
+
+    public CreateUserData() {
+    }
+
```
Можно использовать библиотеку lombok для уменьшения boilerplate кода 
Можно улучшить

pomegranate282922 reviewed (approved):
Отлично, зачет!
